File: Backend/src/main/java/com/hospital/management/doctor/entity/Doctor.java
package com.hospital.management.doctor.entity;

import com.hospital.management.user.entity.User;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "doctors")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Doctor {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String firstName;

    @Column(nullable = false)
    private String lastName;

    @Column(nullable = false, unique = true)
    private String email;

    @Column(nullable = false)
    private String specialization;

    @Column(nullable = false)
    private Double consultationFee;

    private String phone;

    // ✅ IMPORTANT FIX FOR LOGIN MAPPING
    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
}

File: Frontend/lib/modules/doctor/models/doctor.dart
class Doctor {
  final int id;
  final int? userId;
  final String firstName;
  final String lastName;
  final String email;
  final String specialization;
  final double consultationFee;
  final String phone;

  Doctor({
    required this.id,
    this.userId,
    required this.firstName,
    required this.lastName,
    required this.email,
    required this.specialization,
    required this.consultationFee,
    required this.phone,
  });

  // ✅ ADD THIS (fixes your error)
  String get fullName => "$firstName $lastName";

  factory Doctor.fromJson(Map<String, dynamic> json) {
    return Doctor(
      id: json['id'],
      firstName: json['firstName'],
      lastName: json['lastName'],
      email: json['email'],
      specialization: json['specialization'],
      consultationFee: (json['consultationFee'] as num).toDouble(),
      phone: json['phone'],
      userId: json['userId'], // optional safe mapping
    );
  }
}